dot/network/message_cache.go

```diff
@@ -55,6 +56,7 @@ func (m *messageCache) put(peer peer.ID, msg NotificationsMessage) (bool, error)
 func (m *messageCache) exists(peer peer.ID, msg NotificationsMessage) bool {
 	key, err := generateCacheKey(peer, msg)
 	if err != nil {
+		logger.Errorf("could not generate cache key: %s", err)
```

Comment:
can we change the `exists` method to return an `(bool, error)`? and at `generateCacheKey` if the `msg` is a handshake return and error `ErrHandshakeNotSupported`

Reply:
we don't store handshake in `messageCache`. I dont think you would you want to check if message from messageCache is a handshake. Also, messageCache is for out going messages, so we have cntrol over it.

> can we change the exists method to return an (bool, error)?
No other piece of code needs this error. Only thing we are going to do with it, is logging it. We want to keep the code running in case of error here. return false in case of error is enough here,